tcms/core/tests/test_forms.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name
+
+from django.test import TestCase
+
+from tcms.core.forms.fields import UserField
+from tcms.tests.factories import UserFactory
+from django.core.exceptions import ValidationError
+
+
+class TestUserField(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory(username='admin', email='admin@example.com')
+        cls.user.set_password('admin')  # nosec:B106:hardcoded_password_funcarg
+        cls.user.is_superuser = True
+        cls.user.is_staff = True  # enables access to admin panel
+        cls.user.save()
+
+        cls.userField = UserField()
+
+    def test_valid_user(self):
+        formUser = self.userField.clean("admin")
+        self.assertEqual(formUser.email, self.user.email)
+        self.assertEqual(formUser.username, self.user.username)
+
+    def test_empty_username(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.userField.clean('')
+
+    def test_none_username(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.userField.clean(None)
+
+    def test_none_username_not_required(self):
+        self.userField.required = False
+        self.assertIsNone(self.userField.clean(None))
+        self.userField.required = True
+
+    def test_empty_username_not_required(self):
+        self.userField.required = False
+        self.assertIsNone(self.userField.clean(''))
+        self.userField.required = True
+
+    def test_int_instance_username(self):
+        userID = int(self.user.id)
+        formUser = self.userField.clean(userID)
+        self.assertEqual(formUser.email, self.user.email)
+        self.assertEqual(formUser.username, self.user.username)
+
+    def test_int_instance_username_does_not_exist(self):
+        userID = int(self.user.id + 1)
+        with self.assertRaisesRegex(ValidationError, 'Unknown user_id: \"' + str(userID) + '\"'):
+            self.userField.clean(userID)
+
+    def test_digit_username(self):
+        userID = str(self.user.id)
+        formUser = self.userField.clean(userID)
+        self.assertEqual(formUser.email, self.user.email)
+        self.assertEqual(formUser.username, self.user.username)
+
+    def test_digit_username_does_not_exist(self):
+        userID = str(self.user.id + 1)
+        with self.assertRaisesRegex(ValidationError, 'Unknown user_id: \"' + str(userID) + '\"'):
+            self.userField.clean(userID)
+
+    def test_string_username_does_not_exist(self):
+        with self.assertRaisesRegex(ValidationError, 'Unknown user: "UnknownUser"'):
+            self.userField.clean('UnknownUser')
```

Comment:
New line in the end of the file is missing

Reply:
Are you sure about it? This is not reported by CI and i can see it in local and there is no error reported when i check it with flake8.